### PR TITLE
switch to a more TH2D centric framework for the binning

### PR DESCRIPTION
**Pull request by clark2668**
This tries to resolve some binning issue by removing any "custom" bin indexing, and just letting TH2D do all the work.

### CODE REVIEW

File: AraCorrelator/RayTraceCorrelator.cxx [hunk 1]
```diff
@@ -42,38 +42,36 @@ void RayTraceCorrelator::SetAngularConfig(double angularSize){
...
 
-void RayTraceCorrelator::ConvertAngleToBins(double theta, double phi, 
-    int &thetaBin, int &phiBin
-    ){
+int RayTraceCorrelator::ConvertAngleToTH2DGlobalBin(double theta, double phi){
 
-    if(abs(theta) > 91 || isnan(theta)){
+    if(abs(theta) > 90 || isnan(theta)){
         char errorMessage[400];
         sprintf(errorMessage,"Requested theta angle (%e) is not supported. Range should be -91 to 91\n", theta);
```
Comment (abigailbishop) [inline AraCorrelator/RayTraceCorrelator.cxx:63]: Do you need to update this error message to match the change above?

File: AraCorrelator/RayTraceCorrelator.cxx
Comment (abigailbishop) [inline AraCorrelator/RayTraceCorrelator.cxx:1]: Do you need to update this error message to match the change above?

File: AraCorrelator/RayTraceCorrelator.cxx [hunk 2]
```diff
@@ -42,38 +42,36 @@ void RayTraceCorrelator::SetAngularConfig(double angularSize){
...
         throw std::invalid_argument(errorMessage);
     }
 
-    if(abs(phi)>181 || isnan(phi)){
+    if(abs(phi)>180 || isnan(phi)){
         char errorMessage[400];
         sprintf(errorMessage,"Requested phi angle (%e) is not supported. Range should be -181 to 181\n", theta);
```
Comment (abigailbishop) [inline AraCorrelator/RayTraceCorrelator.cxx:69]: Do you need to update this error message to match the change above?

File: AraCorrelator/RayTraceCorrelator.cxx [hunk 3]
```diff
@@ -42,38 +42,36 @@ void RayTraceCorrelator::SetAngularConfig(double angularSize){
...
     }
 }
 
-void RayTraceCorrelator::ConvertAngleToBins(double theta, double phi, 
-    int &thetaBin, int &phiBin
-    ){
+int RayTraceCorrelator::ConvertAngleToTH2DGlobalBin(double theta, double phi){
 
-    if(abs(theta) > 91 || isnan(theta)){
+    if(abs(theta) > 90 || isnan(theta)){
```
Comment (abigailbishop) [inline AraCorrelator/RayTraceCorrelator.cxx:61]: Why is this just >90? Why not >=90?

File: AraCorrelator/RayTraceCorrelator.cxx (hunk 1 above)
Comment (clark2668) [reply AraCorrelator/RayTraceCorrelator.cxx]: Good catch! Done!

File: AraCorrelator/RayTraceCorrelator.cxx (hunk 2 above)
Comment (clark2668) [reply AraCorrelator/RayTraceCorrelator.cxx]: Good catch! Done!

File: AraCorrelator/RayTraceCorrelator.cxx (hunk 3 above)
Comment (clark2668) [reply AraCorrelator/RayTraceCorrelator.cxx:61]: I think ==90 is actually ok? Like, if the upper bin edge is inclusive, then this is okay, right? Or am I underthinking it?

Comment (abigailbishop) [reply AraCorrelator/RayTraceCorrelator.cxx:61]: Oh yes- I misread it. All good here!

Comment (clark2668) [reply AraCorrelator/RayTraceCorrelator.cxx:61]: Okay, thanks!

**Comment by abigailbishop** [comment]
Looks good to me Brian, thanks for doing this! Did you check if this resolves the seg fault I was getting earlier?  If not I can try to in a little bit!

**Comment by marcomuzio** [comment]
Looks good to me! I say merge it.

**Pull request merged**